src/devices/Bmxx80/Bmx280Base.cs

```diff
@@ -0,0 +1,279 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+//Ported from https://github.com/adafruit/Adafruit_BMP280_Library/blob/master/Adafruit_BMP280.cpp
+//Formulas and code examples can also be found in the datasheet http://www.adafruit.com/datasheets/BST-BMP280-DS001-11.pdf
+
+using System;
+using System.Device.I2c;
+using System.IO;
+using System.Threading.Tasks;
+using Iot.Device.Bmxx80.PowerMode;
+using Iot.Device.Bmxx80.Register;
+using Iot.Units;
+
+namespace Iot.Device.Bmxx80
+{
+    /// <summary>
+    /// Represents the core functionality of the Bmx280 family.
+    /// </summary>
+    public abstract class Bmx280Base : Bmxx80Base
+    {
+        /// <summary>
+        /// Default I2C bus address.
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x77;
+
+        /// <summary>
+        /// Secondary I2C bus address.
+        /// </summary>
+        public const byte SecondaryI2cAddress = 0x76;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Bmx280Base"/> class.
+        /// </summary>
+        /// <param name="deviceId">The ID of the device.</param>
+        /// <param name="i2cDevice">The <see cref="I2cDevice"/> to create with.</param>
+        protected Bmx280Base(byte deviceId, I2cDevice i2cDevice)
+            : base(deviceId, i2cDevice)
+        {
+            _controlRegister = (byte)Bmx280Register.CTRL_MEAS;
+            TemperatureFine = int.MinValue;
```

Comment:
Won't this also need to be assigned in Bme680 case? that one doesn't inherit from this base class, and it does use it on Bme680.CompensatePressure. Perhaps you want to initialize this on Bmxx80Base?

Reply:
Thinking about this, do we really want this check for the TemperatureFine value? Ideally, we always want to have a fresh reading of temperature for pressure.
```
            //Read the temperature first to load the t_fine value for compensation
            if (TemperatureFine == int.MinValue)
            {
                await ReadTemperatureAsync();
            }
```
I think the following code is trying to save a call to `ReadTemperatureAsync()` but once we've made that call, this is essentially never true again. Potentially, pressure will use an old temperature reading here which is going to effect the accuracy. I guess it depends on how the sensor class is used in an application (i.e. is it a new instance every time or is it injected in as a singleton) but I think in principle, it would be better to get an up-to-date reading of temperature.

Reply:
I think I agree, and I don't believe it will be a very expensive call, it is just performing another read and that way you will get a more accurate reading of pressure. I believe calling ReadTemperatureAsync always might be better.